arrow-libs/fx/arrow-fx-resilience/src/commonTest/kotlin/arrow/fx/resilience/CircuitBreakerTest.kt

```diff
@@ -1,274 +1,314 @@
+        assertEquals(resetTimeout, s.resetTimeout)
       }
 
-      // After resetTimeout passes, CB should still be Open, and we should be able to reset to Closed.
-      delay(resetTimeout + 10.milliseconds)
+      else -> fail("Invalid state: Expect CircuitBreaker.State.Open but found $s")
+    }
 
-      when (val s = cb.state()) {
-        is CircuitBreaker.State.Open -> {
-          s.resetTimeout shouldBe resetTimeout
-        }
-        else -> fail("Invalid state: Expect CircuitBreaker.State.Open but found $s")
+    assertTrue(cb.state() is CircuitBreaker.State.Open)
+    // If CircuitBreaker is Open our tasks our rejected
+    assertFailsWith<CircuitBreaker.ExecutionRejected> {
+      cb.protectOrThrow { throw dummy }
+    }
+
+    // After resetTimeout passes, CB should still be Open, and we should be able to reset to Closed.
+    timeSource += resetTimeout + 10.milliseconds
+
+    when (val s = cb.state()) {
+      is CircuitBreaker.State.Open -> {
+        assertEquals(resetTimeout, s.resetTimeout)
       }
 
-      val checkHalfOpen = CompletableDeferred<Unit>()
-      val delayProtectLatch = CompletableDeferred<Unit>()
-      val stateAssertionLatch = CompletableDeferred<Unit>()
-
-      @Suppress("DeferredResultUnused")
-      async { // Successful tasks puts circuit breaker back in HalfOpen
-        // Delay protect, to inspect HalfOpen state.
-        Either.catch {
-          cb.protectOrThrow {
-            checkHalfOpen.complete(Unit)
-            delayProtectLatch.await(); throw dummy
-          }
+      else -> fail("Invalid state: Expect CircuitBreaker.State.Open but found $s")
+    }
+
+    val checkHalfOpen = CompletableDeferred<Unit>()
+    val delayProtectLatch = CompletableDeferred<Unit>()
+    val stateAssertionLatch = CompletableDeferred<Unit>()
+
+    @Suppress("DeferredResultUnused")
+    async { // Successful tasks puts circuit breaker back in HalfOpen
+      // Delay protect, to inspect HalfOpen state.
+      Either.catch {
+        cb.protectOrThrow {
+          checkHalfOpen.complete(Unit)
+          delayProtectLatch.await(); throw dummy
         }
-        stateAssertionLatch.complete(Unit)
       }
+      stateAssertionLatch.complete(Unit)
+    }
 
-      checkHalfOpen.await()
+    checkHalfOpen.await()
 
-      when (val s = cb.state()) {
-        is CircuitBreaker.State.HalfOpen -> {
-          s.resetTimeout shouldBe resetTimeout
-        }
-        else -> fail("Invalid state: Expect CircuitBreaker.State.HalfOpen but found $s")
+    when (val s = cb.state()) {
+      is CircuitBreaker.State.HalfOpen -> {
+        assertEquals(resetTimeout, s.resetTimeout)
       }
 
-      // Rejects all other tasks in HalfOpen
-      shouldThrow<CircuitBreaker.ExecutionRejected> { cb.protectOrThrow { throw dummy } }
-      shouldThrow<CircuitBreaker.ExecutionRejected> { cb.protectOrThrow { throw dummy } }
+      else -> fail("Invalid state: Expect CircuitBreaker.State.HalfOpen but found $s")
+    }
```

Comment:
**Nit:** This pattern seems to come back a lot, we can create a small DSL for it. Something like.

```kotlin
inline fun reified <A, reified B: A> assert(expected: A, block: B.(b: B) -> Unit): Unit =
  if(expected is B) block(expected, expected) else fail("Expected ${B::class.simpleName} but found $expected)
```

Reply:
Adding the receiver `B` to the lambda parameter called `block` was causing this assertion to fail:
```
assert(cb.state()) { s: CircuitBreaker.State.Open -> assertEquals(resetTimeout * exponentialBackoffFactor, s.resetTimeout) }
```
The reason is that the assertion was reading the `resetTimeout` and `exponentialBackoffFactor` properties from the `B` value instead of those provided by the test suite.